transcendence-app/src/game/game.queue.service.ts

```diff
@@ -0,0 +1,146 @@
+import { Injectable } from '@nestjs/common';
+import { Server } from 'socket.io';
+import { GameId, UserId } from './infrastructure/db/memoryModels';
+import { IGamesOngoingRepository } from './infrastructure/db/gamesongoing.repository';
+import { IChallengesPendingRepository } from './infrastructure/db/challengespending.repository';
+import {
+  GameChallengeDto,
+  GameChallengeResponseDto,
+  gameQueueServerToClientWsEvents,
+} from 'pong-engine';
+import { GamePairingStatusDto } from './dto/game.pairing.status.dto';
+
+@Injectable()
+export class GameQueueService {
+  public socket: Server | null = null;
+  private gameQueue: Record<GameId, [UserId, null]> | null = null;
```

Comment:
In an earlier version, I did not store the the gameroom entity (as something that has gameId and user(s) id(s), and I didn't make either player join any room, until I had both players matched.

But then, I noticed that I had to duplicate more code, or store the client as it waits (which I'm not sure is safe, since the client may have disconnected since it first joined), since when the queue was full, I had to either send a game ready ws event, to two userIds instead of just one gameroomId, or have a possibly disconnected client join a newly created room.